stdlib/public/runtime/Errors.cpp

```diff
@@ -22,11 +22,80 @@
 #include <pthread.h>
 #include <stdarg.h>
 #include "swift/Runtime/Debug.h"
+#include "swift/Basic/Demangle.h"
+#include "swift/Config.h"
+#include <regex>
+#include <cxxabi.h>
+#include <iostream>
+#include <execinfo.h>
 
 #ifdef __APPLE__
 #include <asl.h>
 #endif
 
+static std::string
+cppDemangle(std::string line, bool *success)
+{
+  std::regex cppRegex("(.*?)(_Z[_a-zA-Z0-9$]+)(.*)");
+  std::smatch matches;
+  std::regex_match(line, matches, cppRegex);
+  if (matches.size() < 4) {
+    return line;
+  }
+  auto str = matches[2].str();
+  int status;
+  char *demangled = abi::__cxa_demangle(str.c_str(), 0, 0, &status);
+  if (demangled == NULL || status != 0) {
+    *success = false;
+    return line;
+  }
+  *success = true;
+  return matches[1].str() + demangled + matches[3].str();
+}
+
+static std::string
+swiftDemangle(std::string line, bool *success)
+{
+  std::regex swiftRegex("(.*?)(_T[_a-zA-Z0-9$]+)(.*)");
```

Comment:
I think an ad-hoc string processing loop would be preferable to pulling in std::regex too. We wouldn't need to depend on possibly-broken libstdc++ that way too, and the pattern here is trivial enough that a regex isn't really buying anything.